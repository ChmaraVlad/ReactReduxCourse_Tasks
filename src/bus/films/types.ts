// Core
import { CaseReducer, PayloadAction } from '@reduxjs/toolkit';

// State
export type Film = {
    title: string,
    episode_id: number,
    opening_crawl: string,
    director: string,
    producer: string,
    release_date: string,
    characters: string[],
    planets: string[],
    starships:string[],
    vehicles: string[],
    species: string[],
    created: string,
    edited: string,
    url: string
}
export type Films = Array<Film>
export type FilmsState = Films | null

// Contracts
export type BaseContact<T = any> = CaseReducer<FilmsState, PayloadAction<T>>
