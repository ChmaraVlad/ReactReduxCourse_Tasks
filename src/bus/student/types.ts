export type Student = {
    firstName: string,
    lastName: string,
    email: string,
    age: string,
    speciality: string,
    sex: string,
}

export type StudentState = {
    results: Student[];
}

export const CREATE_STUDENT = '[STUDENT] - CREATE_STUDENT';
export type createStudentAction = {
    type: typeof CREATE_STUDENT;
    payload: Student
};

export type StudentActionTypes = createStudentAction;
