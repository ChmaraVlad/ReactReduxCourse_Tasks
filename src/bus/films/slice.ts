// Core
import { createSlice } from '@reduxjs/toolkit';

// Types
import * as types from './types';

// Reducers
import * as reducers from './reducers';

const initialState = {
    data:       null,
    isFetching: false,
};

export const filmsSlice = createSlice<types.FilmsState, typeof reducers>({
    name: 'films',
    initialState,
    reducers,
});

export const sliceName = filmsSlice.name;
export const filmsActions = filmsSlice.actions;
export default filmsSlice.reducer;
