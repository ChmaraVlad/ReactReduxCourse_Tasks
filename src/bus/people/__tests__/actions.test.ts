// Actions
import { peopleActions } from '../slice';

// Types
import { ExtendedPerson } from '../types';

const testPerson: ExtendedPerson = {
    name:       'string',
    height:     'string',
    mass:       'string',
    hair_color: 'string',
    skin_color: 'string',
    eye_color:  'string',
    birth_year: 'string',
    gender:     'string',
    homeworld:  'string',
    films:      [],
    species:    [],
    vehicles:   [],
    starships:  [],
    created:    'string',
    edited:     'string',
    url:        'string',
    id:         'string',
};

describe('people actions', ()=>{
    test('should change people Fetching Status to true', ()=>{
        expect(peopleActions.setPeopleFetchingStatus(true)).toMatchSnapshot();
    });
    test('should change people Fetching Status to false', ()=>{
        expect(peopleActions.setPeopleFetchingStatus(false)).toMatchSnapshot();
    });
    test('should fetch people', ()=>{
        expect(peopleActions.setPeople([ testPerson ])).toMatchSnapshot();
    });
    test('should fetch person', ()=>{
        expect(peopleActions.setPerson(testPerson)).toMatchSnapshot();
    });

    // test('should set fetching error', ()=>{
    //     expect(filmsActions.setFetchingError('We have an error !')).toMatchSnapshot();
    // });
});
