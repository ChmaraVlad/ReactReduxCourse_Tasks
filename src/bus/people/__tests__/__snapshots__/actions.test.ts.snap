// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`people actions should change people Fetching Status to false 1`] = `
Object {
  "payload": false,
  "type": "people/setPeopleFetchingStatus",
}
`;

exports[`people actions should change people Fetching Status to true 1`] = `
Object {
  "payload": true,
  "type": "people/setPeopleFetchingStatus",
}
`;

exports[`people actions should fetch people 1`] = `
Object {
  "payload": Array [
    Object {
      "birth_year": "string",
      "created": "string",
      "edited": "string",
      "eye_color": "string",
      "films": Array [],
      "gender": "string",
      "hair_color": "string",
      "height": "string",
      "homeworld": "string",
      "id": "string",
      "mass": "string",
      "name": "string",
      "skin_color": "string",
      "species": Array [],
      "starships": Array [],
      "url": "string",
      "vehicles": Array [],
    },
  ],
  "type": "people/setPeople",
}
`;

exports[`people actions should fetch person 1`] = `
Object {
  "payload": Object {
    "birth_year": "string",
    "created": "string",
    "edited": "string",
    "eye_color": "string",
    "films": Array [],
    "gender": "string",
    "hair_color": "string",
    "height": "string",
    "homeworld": "string",
    "id": "string",
    "mass": "string",
    "name": "string",
    "skin_color": "string",
    "species": Array [],
    "starships": Array [],
    "url": "string",
    "vehicles": Array [],
  },
  "type": "people/setPerson",
}
`;
